src/app/clusters/mode-base-server/mode-base-server.h

```diff
@@ -0,0 +1,215 @@
+     * @param aEndpointId The endpoint on which this cluster exists. This must match the zap configuration.
+     * @param aClusterId The ID of the ModeBase aliased cluster to be instantiated.
+     * @param aFeature The bitmask value that identifies which features are supported by this instance.
+     */
+    Instance(EndpointId aEndpointId, ClusterId aClusterId, uint32_t aFeature) :
+        CommandHandlerInterface(Optional<EndpointId>(aEndpointId), aClusterId),
+        AttributeAccessInterface(Optional<EndpointId>(aEndpointId), aClusterId), mEndpointId(aEndpointId), mClusterId(aClusterId),
+        mFeature(aFeature)
+    {
+        mCurrentMode = 0; // This is a temporary value and may not be valid. We will change this to the value of the first
+                          // mode in the list at the start of the Init function to ensure that it represents a valid mode.
+        mStartUpMode = DataModel::Nullable<uint8_t>(); // Initialised to null
+        mOnMode      = DataModel::Nullable<uint8_t>(); // Initialised to null
+    }
+
+    ~Instance() override;
+
+    template <typename RequestT, typename FuncT>
+    void HandleCommand(HandlerContext & handlerContext, FuncT func);
+
+    // The following functions should be overridden by the SDK user to implement the business logic of their application.
+
+    /**
+     * This init function will be called during the ModeBase server initialization after the Instance information has been
+     * validated and the Instance has been registered. This can be used to initialise app logic.
+     */
+    virtual CHIP_ERROR AppInit() = 0;
+
+    /**
+     * Returns the number of modes managed by this instance.
+     */
+    virtual uint8_t NumberOfModes() = 0;
+
+    /**
+     * Get the mode label of the Nth mode in the list of modes.
+     * @param modeIndex The index of the mode to be returned. It is assumed that modes are indexable from 0 and with no gaps.
+     * @param label A reference to the mutable char span which will be mutated to receive the label on success. Use
+     * CopyCharSpanToMutableCharSpan to copy into the MutableCharSpan.
+     * @return Returns a CHIP_NO_ERROR if there was no error.
+     */
+    virtual CHIP_ERROR GetModeLabelByIndex(uint8_t modeIndex, chip::MutableCharSpan & label) = 0;
+
+    /**
+     * Get the mode value of the Nth mode in the list of modes.
+     * @param modeIndex The index of the mode to be returned. It is assumed that modes are indexable from 0 and with no gaps.
+     * @param value a reference to the uint8_t variable that is to contain the mode value.
+     * @return Returns a CHIP_NO_ERROR if there was no error.
+     */
+    virtual CHIP_ERROR GetModeValueByIndex(uint8_t modeIndex, uint8_t & value) = 0;
+
+    /**
+     * Get the mode tags of the Nth mode in the list of modes.
+     * The caller must make sure the List points to an existing buffer of sufficient size to hold the spec-required number
```

Comment:
This is documentation for our API consumer, which is the entity overriding this, right?

How about:
```suggestion
     * The caller will make sure the List points to an existing buffer of sufficient size to hold the spec-required number
```

Reply:
It does make sense, but, are we then assuming that the only the SDK will make use of this method? That is likely to be correct.